pipestat/pipestat.py

```diff
@@ -146,9 +144,22 @@ def _select_value(
             _, cfg_schema = read_yaml_data(CFG_SCHEMA, "config schema")
             validate(cfg, cfg_schema)
 
-        self[NAME_KEY] = _select_value(
-            "namespace", namespace, self[CONFIG_KEY], env_var=ENV_VARS["namespace"]
+        namespace = _select_value(
+            "namespace",
+            namespace,
+            self[CONFIG_KEY],
+            strict=False,
+            env_var=ENV_VARS["namespace"],
         )
+        if namespace is None:
+            msg = (
+                "Could not determine value for namespace. "
+                "This may be passed directly to the pipestat manager constructor, "
+                "or through a configuration file, "
+                f"or as an environment variable: {ENV_VARS['namespace']}"
+            )
+            raise PipestatError(msg)
+        self[NAME_KEY] = namespace
```

Comment:
Here's the relevant change for #3 -- the check is loosened to `strict=False` to avoid a double-exception traceback, and the error message is tweaked.